packages/react/src/useSecurityKeys.ts

```diff
@@ -0,0 +1,154 @@
+import { useEffect, useState } from 'react'
+
+import {
+  ActionErrorState,
+  ActionSuccessState,
+  AddSecurityKeyHandlerResult,
+  addSecurityKeyPromise,
+  ErrorPayload,
+  OTHER_ERROR_CODE
+} from '@nhost/core'
+
+import { useNhostClient } from './useNhostClient'
+
+interface AddSecurityKeyHandler {
+  (
+    /** Optional human-readable name of the security key */
+    nickname?: string
+  ): Promise<AddSecurityKeyHandlerResult>
+}
+interface RemoveSecurityKeyHandlerResult extends ActionErrorState, ActionSuccessState {}
+interface RemoveSecurityKeyHandler {
+  (
+    /** Unique identifier of the security to remove */
+    id: string
+  ): Promise<RemoveSecurityKeyHandlerResult>
+}
+
+export interface SecurityKey {
+  id: string
+  nickname?: string
+}
+
+export interface SecurityKeysHookResult extends ActionErrorState, ActionSuccessState {
+  /** Add a security key to the current user with the WebAuthn API */
+  add: AddSecurityKeyHandler
+  /**  List the security keys of the current user */
+  list: SecurityKey[]
+  /** Remove the given security key from the list of allowed keys for the current user */
+  remove: RemoveSecurityKeyHandler
+}
+
+interface SecurityKeysHook {
+  (): SecurityKeysHookResult
+}
+
+/**
+ * Use the hook `useSecurityKeys` to list, add and remove the WebAuthn security keys of the user.
+ *
+ * When WebAuthn is enabled, the `add` function will work as expected.
+ *
+ * You have to make sure the current user has correct Hasura permissions on the `auth.user_authenticators` table to list and/or remove their keys:
+ *
+ * - List keys: permissions to select the `id` and `nickname` columns
+ * - Remove keys: permissions to delete rows
+ *
+ * It is recommended to add the custom check `{ user_id: { _eq: "X-Hasura-User-Id" } }` to make sure the user can only list and remove their own keys.
```

Comment:
Is this the first time we use the GraphQL API and rely on permissions to be correct to get data?

As an example, we don't do that for `nhost.auth.getUser()`.

If users have incorrect permissions set for the `auth.user_authenticators` table, then `list` will have incorrect values? Not sure I think that's a good idea.

I think we should either
- Not have `list`, and instead recommend users to use GraphQL to list security keys
- Have a specific endpoint in Hasura Auth for users to get their keys

What do you think?

Reply:
- `nhost.auth.getUser()` only decodes the JWT, which is a bit tricky as it's only updated every 15 minutes. For instance, if the user changed their display name through a mutation, `getUser()` will send the wrong value. This is not something we can or should extend elsewhere
- no permission -> no list
- no list: I think it will make DX overcomplicated to tell "you can add a key, but if you want to list them or to remove one, then you will have to write your own query/mutation, etc etc". It feels really wrong to offer a simple solution to add a key, but not to list/remove them, which should be a really basic thing do
- specific endpoint: "no, god, please, no" (Michael Scott). We have a GraphQL endpoint that does the job, with the two authz layers that are configurable through metadata. I believe we should avoid re-creating a side REST API. GraphQL & Hasura is a perk, not a constraint.

Reply:
I'm just worried that we're introducing a new pattern of managing data. `list` and `remove` is just an abstractios of GraphQL. They are simply graphql queries and mutations under the hood.

We don't have this GraphQL abstraction pattern anywhere else in our SDK.

I feel we should leave GraphQL operations to the developer and their GraphQL client of choice.

Reply:
I understand. We decided to set three APIs and only one GraphQL. From the developer perspective, I would expect not to be forced to juggle across APIs to use simple CRUD feature.

Reply:
We agreed offline to remove `list` and `remove` as helpers in the React SDK because:

- To list and remove security keys can be done with the GraphQL API already.
- We can't have `list` and `remove` in the `@nhost/nhost-js` for technical reasons, which made the SDKs inconsistent.